src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11767,62 +11767,62 @@ private bool CanStartConsequenceExpression()
             ExpressionSyntax expr = this.CurrentToken.Kind switch
             {
                 SyntaxKind.DotToken => _syntaxFactory.MemberBindingExpression(this.EatToken(), this.ParseSimpleName(NameOptions.InExpression)),
                 SyntaxKind.OpenBracketToken => _syntaxFactory.ElementBindingExpression(this.ParseBracketedArgumentList()),
                 _ => throw ExceptionUtilities.Unreachable(),
             };
 
-            while (true)
+            // https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#1288-null-conditional-member-access
+            // Repeatedly consume a single *dependent-access*, such that 'expr' becomes the left operand of 'expandedExpression'.
+            while (tryExpandExpression(expr) is ExpressionSyntax expandedExpression)
+                expr = expandedExpression;
+
+            return expr;
+
+            ExpressionSyntax tryExpandExpression(ExpressionSyntax expr)
             {
-                // Nullable suppression operators should only be consumed by a conditional access
-                // if there are further conditional operations performed after the suppression
-                if (isOptionalExclamationsFollowedByConditionalOperation())
-                {
-                    while (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
-                        expr = _syntaxFactory.PostfixUnaryExpression(SyntaxKind.SuppressNullableWarningExpression, expr, EatToken());
-                }
+                // We should consume suppression '!'s which are in the middle of the consequence, but not at the end.
+                // For example, 'a?.b!.c' should be a cond-access whose RHS is '.b!.c',
+                // while 'a?.b!' should be a suppression-expr containing a cond-access 'a?.b'.
+                using var beforeSuppressionsResetPoint = GetDisposableResetPoint(resetOnDispose: false);
+                while (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+                    expr = _syntaxFactory.PostfixUnaryExpression(SyntaxKind.SuppressNullableWarningExpression, expr, EatToken());
 
                 switch (this.CurrentToken.Kind)
                 {
+                    // a?.b(...)
                     case SyntaxKind.OpenParenToken:
-                        expr = _syntaxFactory.InvocationExpression(expr, this.ParseParenthesizedArgumentList());
-                        continue;
+                        return _syntaxFactory.InvocationExpression(expr, this.ParseParenthesizedArgumentList());
 
+                    // a?.b[...]
                     case SyntaxKind.OpenBracketToken:
-                        expr = _syntaxFactory.ElementAccessExpression(expr, this.ParseBracketedArgumentList());
-                        continue;
+                        return _syntaxFactory.ElementAccessExpression(expr, this.ParseBracketedArgumentList());
 
+                    // a?.b.
                     case SyntaxKind.DotToken:
```

Comment:
i just realized something.  this should likely ensure that this is a `.` token and not a `..` token.  In that case, it would be a range with `a?.b` on the lhs, and something else on the right.  This is likely a preexisting bug we have.  It likely doesn't matter in practice as *semantically* a range cannot have a nullable value on the LHS or RHS.

Reply:
Adding a test